tools/pyboard.py

```diff
@@ -370,6 +370,54 @@ def exec_raw_no_follow(self, command):
         if not data.endswith(b">"):
             raise PyboardError("could not enter raw repl")
 
+        t0 = time.time_ns()
+
+        if True:
+            # raw-paste mode
+            self.serial.write(b"\x05A\x01")
+            data = self.serial.read(2)
+            if data == b"R\x00":
+                # unsupported raw-paste mode
+                pass
+            elif data == b"R\x01":
+                # board supports raw-paste mode
+                data = self.serial.read(2)
+                window_size = data[0] | data[1] << 8
+                window_remain = window_size
+                i = 0
+                while i < len(command_bytes):
+                    while window_remain == 0 or self.serial.inWaiting():
+                        data = self.serial.read(1)
+                        if data == b"\x01":
+                            window_remain += window_size
+                        elif data == b"\x04":
+                            self.serial.write(b"\x04")
+                            return
+                        else:
+                            assert 0, data
+                    b = command_bytes[i : min(i + window_remain, len(command_bytes))]
+                    self.serial.write(b)
+                    window_remain -= len(b)
+                    i += len(b)
+                self.serial.write(b"\x04")
+
+                data = self.read_until(1, b"\x04")
+                if not data.endswith(b"\x04"):
+                    print(data)
+                    raise PyboardError("could not complete raw paste")
+
+                dt = (time.time_ns() - t0) // 1000000 + 1
+                print(
+                    "raw-paste took", dt, len(command_bytes), len(command_bytes) * 10 / (dt * 1e-3)
```

Comment:
Should these time-taken prints only be enabled by some kind of debug / verbose flag?

Reply:
Oops, they need to be removed before merging.  I had them there to compare times with/without raw paste.

Reply:
Ahhh, I wondered if that was the case :+1: